### PR TITLE
 K8SPXC-1198: Remove formatted strings from log messages

### PR DESCRIPTION
**Pull request by inelpandzic**
[K8SPXC-1198](https://jira.percona.com/browse/K8SPXC-1198) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

**CHANGE DESCRIPTION**
---
**Problem:**
In a couple of places, we use formatted log messages, breaking the whole structured logging thing we have.

**Cause:**
Misuse of our logging lib.

**Solution:**
Replace formatted messages with constant messages and proper use of KV pairs for the dynamic part of a log.

**CHECKLIST**
---
**Jira**
- [x] Is the Jira ticket created and referenced properly?
- [x] Does the Jira ticket have the proper statuses for documentation (`Needs Doc`) and QA (`Needs QA`)?
- [x] Does the Jira ticket link to the proper milestone (Fix Version field)?

**Tests**
- [x] Are E2E tests passing?
- [x] Are unit tests passing?
- [x] Are linting tests passing?
- [x] Is an E2E test/test case added for the new feature/change?
- [ ] Are unit tests added where appropriate?
- [ ] Are OpenShift compare files changed for E2E tests (`compare/*-oc.yml`)?

**Config/Logging/Testability**
- [ ] Are all needed new/changed options added to default YAML files?
- [ ] Are the manifests (crd/bundle) regenerated if needed?
- [ ] Did we add proper logging messages for operator actions?
- [ ] Did we ensure compatibility with the previous version or cluster upgrade process?
- [ ] Does the change support oldest and newest supported PXC version?
- [ ] Does the change support oldest and newest supported Kubernetes version?

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| We run 0 out of 36|

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1337/commits/eec7e4315ebc5f90f1ea53ca175e45401858bb9f
image: `perconalab/percona-xtradb-cluster-operator:PR-1337-eec7e431`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| We run 0 out of 36|

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1337/commits/eec7e4315ebc5f90f1ea53ca175e45401858bb9f
image: `perconalab/percona-xtradb-cluster-operator:PR-1337-eec7e431`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| affinity-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/affinity-8-0.log) |
| auto-tuning-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/auto-tuning-8-0.log) |
| cross-site-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/cross-site-8-0.log) |
| demand-backup-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/demand-backup-8-0.log) |
| demand-backup-cloud-8-0 | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/demand-backup-cloud-8-0.log) |
| demand-backup-encrypted-with-tls-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/demand-backup-encrypted-with-tls-8-0.log) |
| haproxy-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/haproxy-5-7.log) |
| haproxy-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/haproxy-8-0.log) |
| init-deploy-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/init-deploy-5-7.log) |
| init-deploy-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/init-deploy-8-0.log) |
| limits-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/limits-8-0.log) |
| monitoring-2-0-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/monitoring-2-0-8-0.log) |
| one-pod-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/one-pod-5-7.log) |
| one-pod-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/one-pod-8-0.log) |
| pitr-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/pitr-8-0.log) |
| proxy-protocol-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/proxy-protocol-8-0.log) |
| proxysql-sidecar-res-limits-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/proxysql-sidecar-res-limits-8-0.log) |
| recreate-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/recreate-8-0.log) |
| restore-to-encrypted-cluster-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/restore-to-encrypted-cluster-8-0.log) |
| scaling-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/scaling-8-0.log) |
| scaling-proxysql-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/scaling-proxysql-8-0.log) |
| scheduled-backup-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/scheduled-backup-5-7.log) |
| scheduled-backup-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/scheduled-backup-8-0.log) |
| security-context-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/security-context-8-0.log) |
| smart-update1-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/smart-update1-8-0.log) |
| smart-update2-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/smart-update2-8-0.log) |
| storage-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/storage-8-0.log) |
| tls-issue-cert-manager-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/tls-issue-cert-manager-8-0.log) |
| tls-issue-cert-manager-ref-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/tls-issue-cert-manager-ref-8-0.log) |
| tls-issue-self-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/tls-issue-self-8-0.log) |
| upgrade-consistency-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/upgrade-consistency-8-0.log) |
| upgrade-haproxy-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/upgrade-haproxy-8-0.log) |
| upgrade-proxysql-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/upgrade-proxysql-8-0.log) |
| users-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/users-5-7.log) |
| users-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/users-8-0.log) |
| validation-hook-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/689f84e5/validation-hook-8-0.log) |
| We run 36 out of 36|

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1337/commits/689f84e5dfe6fc04739bae003f6066eff59ac7ee
image: `perconalab/percona-xtradb-cluster-operator:PR-1337-689f84e5`

File: pkg/controller/pxc/users_without_dp.go [hunk 1]
```diff
@@ -339,28 +339,28 @@ func (r *ReconcilePerconaXtraDBCluster) handleXtrabackupUserWithoutDP(cr *api.Pe
 		return nil
 	}
 
-	logger.Info(fmt.Sprintf("User %s: password changed, updating user", user.Name))
+	log.Info(fmt.Sprintf("User %s: password changed, updating user", user.Name))
```
Comment (egegunes) [inline pkg/controller/pxc/users_without_dp.go:342]: any reason to keep this formatted string?

File: pkg/controller/pxc/users_without_dp.go [hunk 2]
```diff
@@ -391,28 +391,28 @@ func (r *ReconcilePerconaXtraDBCluster) handleReplicationUserWithoutDP(cr *api.P
...
```
Comment (egegunes) [inline pkg/controller/pxc/users_without_dp.go:394]: any reason to keep this formatted string?

File: pkg/controller/pxc/users_without_dp.go [hunk 3]
```diff
@@ -431,21 +431,21 @@ func (r *ReconcilePerconaXtraDBCluster) handleProxyadminUserWithoutDP(cr *api.Pe
...
 
 	err := r.updateProxyUser(cr, internalSecrets, user)
 	if err != nil {
 		return errors.Wrap(err, "update Proxy users")
 	}
-	logger.Info(fmt.Sprintf("User %s: proxy user updated", user.Name))
+	log.Info(fmt.Sprintf("User %s: proxy user updated", user.Name))
```
[the remark this answers is not in the document]
Comment (inelpandzic) [reply pkg/controller/pxc/users_without_dp.go:440]: Thanks, no reason whatsoever, I missed them. Fixing them

[the remark this answers is not in the document]
Comment (inelpandzic) [reply pkg/controller/pxc/users_without_dp.go]: [56a958b](https://github.com/percona/percona-xtradb-cluster-operator/pull/1337/commits/56a958b7833f4ff72ff3444dd210f258adafee0e)

File: pkg/controller/pxc/users_without_dp.go [hunk 4]
```diff
@@ -391,28 +391,28 @@ func (r *ReconcilePerconaXtraDBCluster) handleReplicationUserWithoutDP(cr *api.P
...
 
-	logger.Info(fmt.Sprintf("User %s: password changed, updating user", user.Name))
+	log.Info(fmt.Sprintf("User %s: password changed, updating user", user.Name))
 
 	err := r.updateUserPassWithoutDP(cr, secrets, internalSecrets, user)
 	if err != nil {
 		return errors.Wrap(err, "update replication users pass")
 	}
-	logger.Info(fmt.Sprintf("User %s: password updated", user.Name))
+	log.Info(fmt.Sprintf("User %s: password updated", user.Name))
```
[the remark this answers is not in the document]
Comment (inelpandzic) [reply pkg/controller/pxc/users_without_dp.go]: Fixed

File: pkg/controller/pxc/users_without_dp.go (hunk 2 above)
Comment (inelpandzic) [reply pkg/controller/pxc/users_without_dp.go]: Fixed

File: pkg/controller/pxc/users_without_dp.go [hunk 5]
```diff
@@ -339,28 +339,28 @@ func (r *ReconcilePerconaXtraDBCluster) handleXtrabackupUserWithoutDP(cr *api.Pe
...
 
-	logger.Info(fmt.Sprintf("User %s: password changed, updating user", user.Name))
+	log.Info(fmt.Sprintf("User %s: password changed, updating user", user.Name))
 
 	err := r.updateUserPassWithoutDP(cr, secrets, internalSecrets, user)
 	if err != nil {
 		return errors.Wrap(err, "update xtrabackup users pass")
 	}
-	logger.Info(fmt.Sprintf("User %s: password updated", user.Name))
+	log.Info(fmt.Sprintf("User %s: password updated", user.Name))
```
[the remark this answers is not in the document]
Comment (inelpandzic) [reply pkg/controller/pxc/users_without_dp.go]: Fixed

File: pkg/controller/pxc/users_without_dp.go (hunk 1 above)
Comment (inelpandzic) [reply pkg/controller/pxc/users_without_dp.go]: Fixed

File: pkg/controller/pxc/users_without_dp.go [hunk 6]
```diff
@@ -289,27 +289,27 @@ func (r *ReconcilePerconaXtraDBCluster) handleClustercheckUserWithoutDP(cr *api.
...
 
-	logger.Info(fmt.Sprintf("User %s: password changed, updating user", user.Name))
+	log.Info(fmt.Sprintf("User %s: password changed, updating user", user.Name))
 
 	err := r.updateUserPassWithoutDP(cr, secrets, internalSecrets, user)
 	if err != nil {
 		return errors.Wrap(err, "update clustercheck users pass")
 	}
-	logger.Info(fmt.Sprintf("User %s: password updated", user.Name))
+	log.Info(fmt.Sprintf("User %s: password updated", user.Name))
```
[the remark this answers is not in the document]
Comment (inelpandzic) [reply pkg/controller/pxc/users_without_dp.go]: Fixed

File: pkg/controller/pxc/users_without_dp.go [hunk 7]
```diff
@@ -289,27 +289,27 @@ func (r *ReconcilePerconaXtraDBCluster) handleClustercheckUserWithoutDP(cr *api.
 		return nil
 	}
 
-	logger.Info(fmt.Sprintf("User %s: password changed, updating user", user.Name))
+	log.Info(fmt.Sprintf("User %s: password changed, updating user", user.Name))
```
[the remark this answers is not in the document]
Comment (inelpandzic) [reply pkg/controller/pxc/users_without_dp.go]: Fixed

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| affinity-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/affinity-8-0.log) |
| auto-tuning-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/auto-tuning-8-0.log) |
| cross-site-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/cross-site-8-0.log) |
| demand-backup-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/demand-backup-8-0.log) |
| demand-backup-cloud-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/demand-backup-cloud-8-0.log) |
| demand-backup-encrypted-with-tls-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/demand-backup-encrypted-with-tls-8-0.log) |
| haproxy-5-7 | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/haproxy-5-7.log) |
| haproxy-8-0 | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/haproxy-8-0.log) |
| init-deploy-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/init-deploy-5-7.log) |
| init-deploy-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/init-deploy-8-0.log) |
| limits-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/limits-8-0.log) |
| monitoring-2-0-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/monitoring-2-0-8-0.log) |
| one-pod-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/one-pod-5-7.log) |
| one-pod-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/one-pod-8-0.log) |
| pitr-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/pitr-8-0.log) |
| proxy-protocol-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/proxy-protocol-8-0.log) |
| proxysql-sidecar-res-limits-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/proxysql-sidecar-res-limits-8-0.log) |
| recreate-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/recreate-8-0.log) |
| restore-to-encrypted-cluster-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/restore-to-encrypted-cluster-8-0.log) |
| scaling-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/scaling-8-0.log) |
| scaling-proxysql-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/scaling-proxysql-8-0.log) |
| scheduled-backup-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/scheduled-backup-5-7.log) |
| scheduled-backup-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/scheduled-backup-8-0.log) |
| security-context-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/security-context-8-0.log) |
| smart-update1-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/smart-update1-8-0.log) |
| smart-update2-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/smart-update2-8-0.log) |
| storage-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/storage-8-0.log) |
| tls-issue-cert-manager-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/tls-issue-cert-manager-8-0.log) |
| tls-issue-cert-manager-ref-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/tls-issue-cert-manager-ref-8-0.log) |
| tls-issue-self-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/tls-issue-self-8-0.log) |
| upgrade-consistency-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/upgrade-consistency-8-0.log) |
| upgrade-haproxy-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/upgrade-haproxy-8-0.log) |
| upgrade-proxysql-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/upgrade-proxysql-8-0.log) |
| users-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/users-5-7.log) |
| users-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/users-8-0.log) |
| validation-hook-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1337/56a958b7/validation-hook-8-0.log) |
| We run 36 out of 36|

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1337/commits/56a958b7833f4ff72ff3444dd210f258adafee0e
image: `perconalab/percona-xtradb-cluster-operator:PR-1337-56a958b7`

**Review by hors — APPROVED** [review]

**Review by pooknull — APPROVED** [review]

**Review by egegunes — APPROVED** [review]

**Pull request merged**